### PR TITLE
Allow a custom server URL

### LINKED ISSUE #15: Private Server
We have a private, self hosted Hipchat-Server.

Please add a Parameter to set the base URL like

```
notify:
  hipchat:
    host: hipchat.company.com
    secured: true #default (SSL)
    auth_token: xxxxxxxxxx
    room_id_or_name: 42
    notify: false
```

the call is then: `https://hipchat.company.com/v2/room/42/...`

### PR DESCRIPTION
**Pull request by tboerger**
In order to allow notifications through self-hosted HipChat servers I
have integrated a new option for a custom domain which falls back to the
SaaS version of HipChat.

Fixes #15

### CODE REVIEW

**Comment by davidak** [comment]
that was quick. you are amazing!

sadly i can't review it.

is this the right order? https://github.com/drone-plugins/drone-hipchat/commit/9721d0bbae7a5a1e16839faf935c2b68c63387c8#diff-b2a2fb355215662156774bb6e471b72bR36
i don't understand if that should build the full url. then it should start with url.

**Comment by tboerger** [comment]
> is this the right order? 9721d0b#diff-b2a2fb355215662156774bb6e471b72bR36
> i don't understand if that should build the full url. then it should start with url.

Look at https://github.com/drone-plugins/drone-hipchat/commit/9721d0bbae7a5a1e16839faf935c2b68c63387c8#diff-b2a2fb355215662156774bb6e471b72bR13 :)

File: hipchat.go [hunk 1]
```diff
@@ -8,7 +8,10 @@ import (
 	"net/http"
 )
 
-var notifyURL = "https://api.hipchat.com/v2/room/%s/notification?auth_token=%s"
+var (
+	defaultUrl = "https://api.hipchat.com"
```
Comment (jackspirou) [inline hipchat.go]: `Url` should be `URL` https://github.com/golang/go/wiki/CodeReviewComments#initialisms

File: main.go [hunk 2]
```diff
@@ -32,7 +32,11 @@ func main() {
 		vargs.Template = defaultTemplate
 	}
 
-	client := NewClient(vargs.Room.String(), vargs.Token)
+	client := NewClient(
+		vargs.Url,
```
Comment (jackspirou) [inline main.go]: `Url` should be `URL` https://github.com/golang/go/wiki/CodeReviewComments#initialisms

File: types.go [hunk 3]
```diff
@@ -7,6 +7,7 @@ import (
 // Params are the parameters that the HipChat plugin can parse.
 type Params struct {
 	Notify   bool            `json:"notify"`
+	Url      string          `json:"url"`
```
Comment (jackspirou) [inline types.go]: `Url` should be `URL` https://github.com/golang/go/wiki/CodeReviewComments#initialisms

**Comment by jackspirou** [comment]
Commented on tiny nit, otherwise LGTM

**Comment by tboerger** [comment]
Thanks, will fix and merge it tomorrow

**Comment by tboerger** [comment]
LGTM

**Pull request merged**